src/common/components/header/index.js

```diff
@@ -7,7 +7,7 @@ import styles from './header.css';
 
 const wordmark = 'https://assets.ubuntu.com/v1/d45097a4-snapcraft.io-logotype.svg';
 
-class Header extends Component {
+export class Header extends Component {
```

Comment:
Maybe the issue here is that we have both presentional and connected component in one file.

Because that's what we do in most cases (where we do bind data):

```js
class SomeComponent extends Component {
  render() {
  }
}

SomeComponent.propTypes = {
}

const mapStateToProps = () => {
}

export default connect(mapStateToProps)(SomeComponent);
```

With such pattern we have a source code of pure react component `SomeComponent`, but we export only the bound one - because that's the one used by the app. These are 2 different components, but only one exported.

If we export both we end up with "two separate public interfaces", but at least it makes it possible to test the pure React one.
The other solution will be to split these into 2 files. One for pure react component, so we can easily import and test it. Another one for data binding only if needed.

Sure, in some cases (like with Header) it's fine to pass couple of props, but in more complicated cases it may become troublesome.

If we take the dashboard, so we really want the top container to handle all the data?
Does dashboard need to know about all the data it's child components need? Or should it only serve as a layout for components that handle data on their own?

We need to find some middle ground here I guess. Not exporting pure React components will likely result in not testing them. I don't know what test setup would be needed to test the connected one (mocking the store and passing it to connect), but it will make testing components much harder than it is for pure React components. And I don't think we should make all our components just pure components and leave data binding to parent container, because it will make containers responsible for all data of their child components which I don't think is good and necessary.